src/distrs/beta.jl

```diff
@@ -1,19 +1,71 @@
 # functions related to beta distributions
 
+using HypergeometricFunctions: _₂F₁
+
 import .RFunctions:
-    betapdf,
-    betalogpdf,
-    betacdf,
-    betaccdf,
-    betalogcdf,
-    betalogccdf,
-    betainvcdf,
-    betainvccdf,
+    # betapdf,
+    # betalogpdf,
+    # betacdf,
+    # betaccdf,
+    # betalogcdf,
+    # betalogccdf,
+    # betainvcdf,
+    # betainvccdf,
     betainvlogcdf,
     betainvlogccdf
 
 # pdf for numbers with generic types
-betapdf(α::Real, β::Real, x::Number) = x^(α - 1) * (1 - x)^(β - 1) / beta(α, β)
+betapdf(α::Real, β::Real, x::Real) = exp(betalogpdf(α, β, x))
 
 # logpdf for numbers with generic types
-betalogpdf(α::Real, β::Real, x::Number) = (α - 1) * log(x) + (β - 1) * log1p(-x) - logbeta(α, β)
+function betalogpdf(α::T, β::T, x::T) where T<:Real
+    return if x <= 0.5
```

Comment:
Maybe one could explain the motivation for why this is only used for ``x <= 0.5``?

Reply:
betalogpdf should return -Inf outside of (0,1)
```
    return if (x < zero(T) || x > one(T)) # is (x < 1 | x > 1) better?
        -Inf
    elseif x <= 0.5
```

Reply:
This would not be type stable. Maybe one could just perform computations with `clamp(x, 0, 1)` and return `oftype(result, -Inf)` if `x` is outside the unit interval.

Reply:
Due to the check `x <= 0.5`, it would even be sufficient to perform computations with `max(0, x)`.

Reply:
Why is it not type stable? `typeof(-Inf)` is `Float64`

Reply:
Why is it not type stable? `typeof(-Inf)` is `Float64`
PS. I started a review by mistake, not sure how to remove myself from the reviewer list or dismiss my review -- sorry!

Reply:
> Why is it not type stable?

The other branches might not return a result of type `Float64` (e.g., if the inputs are of type `Float32`).

Reply:
Good point. 

`-T(Inf)` then?

Reply:
No, this is incorrect and will error if the inputs are e.g. of type `Rational`.

Reply:
Right, I am too narrowly focused on my own use case -- some day I will become less of a newb :) 

Thanks for the comments.